Source/Core/Duality/Utility/Coroutines/WaitCondition.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Duality.Utility.Pooling;
+
+namespace Duality
+{
+	public interface IWaitCondition
+	{
+		bool Update();
+	}
+
+	public class WaitCondition : IWaitCondition
+	{
+		private readonly Func<bool> exitCondition;
+
+		public WaitCondition(Func<bool> condition)
+		{
+			this.exitCondition = condition;
+		}
+
+		public bool Update()
+		{
+			return this.exitCondition();
+		}
+	}
+
+	public class WaitCondition<T> : IWaitCondition
+	{
+		private readonly Func<T, T> update;
+		private readonly Func<T, bool> exitCondition;
+		private readonly T startingValue;
+		private T currentValue;
+
+		public WaitCondition(Func<T, T> update, Func<T, bool> exitCondition, T startingValue)
+		{
+			this.update = update;
+			this.exitCondition = exitCondition;
+			this.currentValue = this.startingValue = startingValue;
+		}
+
+		public IWaitCondition Reset()
```

Comment:
Is the reset still being used now that we cannot use Enumerable Reset?

Reply:
Yeah, currently pondering if it's just better to remove it and call it a day. Actually never used it once in my project tbh 😄 I think we can remove it for now, see if it gets requested later on, and decide how to deal with that if the day comes.